File: android/app/src/main/kotlin/com/menumaker/data/remote/models/DishModels.kt
package com.menumaker.data.remote.models

import com.google.gson.annotations.SerializedName

data class DishDto(
    @SerializedName("id") val id: String,
    @SerializedName("business_id") val businessId: String,
    @SerializedName("name") val name: String,
    @SerializedName("description") val description: String?,
    @SerializedName("price_cents") val priceCents: Int,
    @SerializedName("image_url") val imageUrl: String?,
    @SerializedName("category") val category: String?,
    @SerializedName("is_vegetarian") val isVegetarian: Boolean,
    @SerializedName("is_available") val isAvailable: Boolean,
    @SerializedName("created_at") val createdAt: String,
    @SerializedName("updated_at") val updatedAt: String
)

data class DishResponse(
    @SerializedName("success") val success: Boolean,
    @SerializedName("data") val data: DishData
)

data class DishData(
    @SerializedName("dish") val dish: DishDto
)

data class DishListResponse(
    @SerializedName("success") val success: Boolean,
    @SerializedName("data") val data: DishListData
)

data class DishListData(
    @SerializedName("dishes") val dishes: List<DishDto>
)

File: ios/MenuMaker/Data/Models/DishModels.swift
import Foundation

// MARK: - Dish Models

struct Dish: Codable, Identifiable {
    let id: String
    let businessId: String
    let name: String
    let description: String?
    let priceCents: Int
    let imageUrl: String?
    let category: String?
    let isVegetarian: Bool
    let isAvailable: Bool
    let createdAt: String
    let updatedAt: String

    var price: Double {
        Double(priceCents) / 100.0
    }

    var formattedPrice: String {
        String(format: "₹%.2f", price)
    }

    var displayDescription: String {
        description ?? "No description available"
    }

    var displayCategory: String {
        category ?? "Uncategorized"
    }

    var vegetarianBadge: String {
        isVegetarian ? "🌱" : "🍖"
    }
}

struct DishResponse: Decodable {
    let success: Bool
    let data: DishData
}

struct DishData: Decodable {
    let dish: Dish
}

struct DishListResponse: Decodable {
    let success: Bool
    let data: DishListData
}

struct DishListData: Decodable {
    let dishes: [Dish]
}

struct CreateDishRequest: Encodable {
    let businessId: String
    let name: String
    let description: String?
    let priceCents: Int
    let imageUrl: String?
    let category: String?
    let isVegetarian: Bool
    let isAvailable: Bool
}

struct UpdateDishRequest: Encodable {
    let name: String?
    let description: String?
    let priceCents: Int?
    let imageUrl: String?
    let category: String?
    let isVegetarian: Bool?
    let isAvailable: Bool?
}
